sklearn/metrics/pairwise.py

```diff
@@ -933,6 +940,9 @@ def manhattan_distances(X, Y=None, *, sum_over_features=True):
     """
     X, Y = check_pairwise_arrays(X, Y)
 
+    if sum_over_features and PairwiseDistances.is_usable_for(X, Y, metric="manhattan"):
+        return PairwiseDistances.compute(X, Y, metric="manhattan")
```

Comment:
`_sparse_manhattan` is still needed because it allow supporting sparse matrices that aren't CSR by converting them to this format. Such conversation is currently not done for case for `PairwiseDistancesReduction` which only accepts a certain kind of CSR matrices:

https://github.com/scikit-learn/scikit-learn/blob/53d6c07f072b5cfdc4f1e687243f916555dbf174/sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py#L105-L141

I think the removal of `_sparse_manhattan` has to be postponed because as might we first need another PR broaden the kind of supported sparse matrices for `PairwiseDistancesReduction`.

Reply:
But it's not ``_sparse_manhattan`` that does the conversion. In ``manhattan_distances`` we check if one is sparse and if so, we convert both to CSR, then call ``_sparse_manhattan``. So what I meant is that we can't make this check/conversion first and then call PairwiseDistancesReduction instead of ``_sparse_manhattan``

Reply:
Ah yes, 259edc1d64660262ebfa54101d64a474fc52666d treats your remarks.